src/SeleniumTestCase.php

```diff
@@ -3,51 +3,65 @@
 namespace Modelizer\Selenium;
 
 use Modelizer\Selenium\Services\Application as Laravel;
+use Modelizer\Selenium\Services\InteractsWithPage as Interactions;
+use Modelizer\Selenium\Services\WaitsForElements;
+use Modelizer\Selenium\Services\WorksWithDatabases;
 use PHPUnit_Extensions_Selenium2TestCase;
 
 class SeleniumTestCase extends PHPUnit_Extensions_Selenium2TestCase
 {
-    use Laravel;
+    use Laravel,
+        Interactions,
+        WorksWithDatabases,
+        WaitsForElements;
+
+    /**
+     * @var string
+     */
+    protected $baseUrl;
 
     protected function setUp()
     {
         $this->setUpLaravel();
-
-        $this->setBrowserUrl(env('APP_URL', 'http://localhost/'));
+        $this->baseUrl = env('APP_URL', 'http://localhost/');
+        $this->setBrowserUrl($this->baseUrl);
         $this->setBrowser(env('DEFAULT_BROWSER', 'chrome'));
     }
 
-    protected function visit($path)
-    {
-        $this->url($path);
-
-        return $this;
-    }
-
-    protected function see($text, $tag = 'html')
+    public function setupPage()
     {
-        $this->assertContains($text, $this->byTag($tag)->text());
-
-        return $this;
+        if (!empty(getenv('SELENIUM_WIDTH')) && !empty(getenv('SELENIUM_HEIGHT')) &&
+            is_int(intval(getenv('SELENIUM_WIDTH'))) && is_int(intval(getenv('SELENIUM_HEIGHT')))) {
+            $this->prepareSession()->currentWindow()->size([
+                'width'  => intval(getenv('SELENIUM_WIDTH')),
+                'height' => intval(getenv('SELENIUM_HEIGHT')),
+            ]);
+        }
     }
 
-    protected function hold($seconds)
+    /**
+     * Force selenium to wait.
+     *
+     * @param int|float $seconds The number of seconds or partial seconds to wait
+     *
+     * @return $this
+     */
+    protected function wait($seconds = 1)
     {
-        sleep($seconds);
+        usleep($seconds * 1000000);
 
         return $this;
     }
 
-    protected function submitForm($inputs, $selector)
+    /**
+     * Alias for wait.
+     *
+     * @param int $seconds
+     *
+     * @return SeleniumTestCase
+     */
+    public function hold($seconds = 1)
     {
-        $form = $this->byCssSelector($selector);
-
-        foreach ($inputs as $input => $value) {
-            $form->byName($input)->value($value);
-        }
-
-        $form->submit();
-
-        return $this;
+        return $this->wait($seconds);
```

Comment:
Yes, agreed but we don't know how much do ajax call can take. If it's more then 5 secs and we specify for 3 sec in our test case then the test will fail. Take a look(Fast forward it if required, video is lengthy) https://laracasts.com/series/intuitive-integration-testing/episodes/5